src/formatDate.js

```diff
@@ -1,14 +1,77 @@
 'use strict';
 
-/**
- * @param {string} date
- * @param {string[]} fromFormat
- * @param {string[]} toFormat
- *
- * @returns {string}
- */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const arrayDate = date.split(fromFormat[3]);
+  const oldFormat = {};
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    for (let partOfDate = i; partOfDate < arrayDate.length; partOfDate++) {
+      oldFormat[fromFormat[i]] = arrayDate[partOfDate];
+      break;
+    }
+  }
+
+  const oldFormatKeys = Object.keys(oldFormat);
+  let yearOldFormat = '';
+
+  for (const key of oldFormatKeys) {
+    if (key.includes('Y')) {
+      yearOldFormat += key;
+      break;
+    }
+  }
+
+  for (let n = 0; n < toFormat.length - 1; n++) {
+    if (toFormat[n].includes('Y')) {
+      if (toFormat[n] === 'YYYY') {
+        if (toFormat[n] === yearOldFormat) {
```

Comment:
The condition `toFormat[n] === yearOldFormat` assumes the year format in the input matches the expected format. Ensure that the input date is validated against `fromFormat` before processing.